packages/playwright-core/src/client/network.ts

```diff
@@ -703,7 +701,7 @@ export class RouteHandler {
     // Note that context.route handler may be later invoked on a different page,
     // so we only swallow errors for the current page's routes.
     const handlerActivations = page ? this._activeInvocations.get(page) : [...this._activeInvocations.values()];
-    if (this.noWaitOnUnrouteOrClose || noWait) {
+    if (noWait) {
```

Comment:
nit: let's rename `noWait` into `behavior: 'wait' | 'ignoreErrors'` to be consistent?